logserver/src/main/java/alluxio/logserver/AlluxioLogServerProcess.java

```diff
@@ -0,0 +1,307 @@
+    } else if (loggerType.contains("WORKER")) {
+      logFilePath += ("/worker_logs/" + key + ".worker.log");
+    } else {
+      // Should not reach here
+      throw new IllegalStateException("Unknown logger type");
+    }
+    properties.setProperty("log4j.rootLogger", "INFO," + loggerType);
+    properties.setProperty("log4j.appender." + loggerType + ".File", logFilePath);
+    new PropertyConfigurator().doConfigure(properties, clientHierarchy);
+    return clientHierarchy;
+  }
+
+  /**
+   * Runnable object that serves logging clients, i.e. Alluxio servers in this case.
+   */
+  private class LogReceiver implements Runnable {
+    private ServerSocket mServerSocket;
+    private final int mPort;
+    private final String mLoggerType;
+    private Map<InetAddress, LoggerRepository> mInetAddressHashMap;
+
+    /**
+     * Construct {@link LogReceiver} instance.
+     *
+     * @param port port number to bind
+     * @param loggerType string representation of the type of the appender,
+     *                   e.g. MASTER_LOGGER, WORKER_LOGGER, etc
+     */
+    public LogReceiver(int port, String loggerType) {
+      mPort = port;
+      mLoggerType = loggerType;
+      mInetAddressHashMap = new HashMap<>();
+    }
+
+    public void close() {
+      if (mServerSocket != null) {
+        try {
+          mServerSocket.close();
+        } catch (IOException e) {
+          return;
+        }
+      }
+    }
+
+    @Override
+    public void run() {
+      try {
+        mServerSocket = new ServerSocket(mPort);
+      } catch (IOException e) {
+        LOG.error("Failed to bind to port {}.", mPort);
+        return;
+      }
+      int failureCount = 0;
+      while (!mStopped) {
+        try {
+          Socket client = mServerSocket.accept();
+          InetAddress inetAddress = client.getInetAddress();
+          // Currently loggerRepository is always null because we are not using the map.
+          LoggerRepository loggerRepository = mInetAddressHashMap.get(inetAddress);
+          if (loggerRepository == null) {
+            try {
+              loggerRepository = configureHierarchy(inetAddress, mLoggerType);
+              mInetAddressHashMap.put(inetAddress, loggerRepository);
+            } catch (URISyntaxException e) {
+              LOG.warn("URI syntax error.");
+              continue;
+            }
+          }
+          AlluxioLog4jSocketNode clientSocketNode =
+              new AlluxioLog4jSocketNode(client, loggerRepository);
+          int retryCount = 0;
+          long remainTimeInMillis = mRequestTimeoutInMillis;
+          while (true) {
+            try {
+              mThreadPool.execute(clientSocketNode);
+              break;
+            } catch (Throwable t) {
+              if (t instanceof RejectedExecutionException) {
+                retryCount++;
+                try {
+                  if (remainTimeInMillis > 0) {
+                    long sleepTimeInMillis = ((long) (mRandom.nextDouble()
```

Comment:
Is there anything here: https://github.com/Alluxio/alluxio/tree/master/core/common/src/main/java/alluxio/retry that will do something similar?

Reply:
Done.